LSPS5/README.md

```diff
@@ -0,0 +1,264 @@
+    of the reason codes defined below.
+  * `pushed_at` is a required datetime, which is the time
+    at the LSP when it generated this push request.
+  * `zbase` is a required signature from the LSP, signing
+    the message defined below.
+
+For `version` 1 `zbase` signatures, the signature signs
+the following JSON string:
+
+    "LSPS5:Version 1: Reason ${reason} at time ${pushed_at}."
+
+Where `${reason}` is the shortest decimal text form of the
+`reason` code, and `${pushed_at}` is the ISO8601 format
+`YYYY-mm-ddThh:mm:ss.uuuZ` of the `pushed_at` datetime.
+
+For example, for the above `version` 1 `details` object,
+the JSON string to be signed would be `"LSPS5:Version 1:
+Reason 1 at time 2023-04-25T14:27:03.842Z."`
+
+For `version` 1, the following `reason` codes are defined:
+
+* `1` - The client has a pending incoming payment.
+* `2` - The LSP wants to recover some of the liquidity from
+  the client.
+* `3` - The LSP wants to change feerates for channels or
+  other management actions.
+* `4` - The LSP wants to end services to the client.
+
+The client application server, on receiving a `POST`
+request on its URI:
+
+* MUST check the `GET` parameters of the URI for any
+  authorization that the specified `lsp` node ID has
+  been authorized to make the push notification.
+* MUST check the `GET` parameters of the URI for which
+  instance of the client application the notification
+  is for.
+* for `version` 1, MUST validate that the `details.zbase`
+  signs the message specified, based on `details.reason`
+  and `details.pushed_at`.
+* for `version` 1, MUST check if the given `lsp` has
+  given a `pushed_at` in a previous `POST`, and if so,
+  MUST check that the current `pushed_at` is strictly
+  later than the latest previous `pushed_at`; if not, it
+  MUST ignore the push notification.
+* MAY filter out push notifications by other policies,
+  such as too many push notifications in a time frame,
+  or rejecting particular `reason` codes.
+
+> **Rationale** The `pushed_at` check is intended to
+> avoid replay attacks, where some third party is able
+> to copy the JSON object sent by the LSP and replays
+> it in order to frame the LSP for spamming the client.
+
+The `POST` response is unspecified.
+The LSP MUST ignore the response to the `POST`.
+
+> **Rationale** As a notification, the user of the
+> client may ignore it, or disable notifications from
+> the client application entirely; thus, delivery of
+> the notification is never assured and it is pointless
+> to feed back anything to the LSP about whether the
+> notification was delivered to the client application
+> server, the mobile OS server, or the client mobile
+> device.
+
+If the client application server accepts the push
+notification, it MUST construct a suitable message to
+display to the user, and send that message, together
+with any required authentication and authorization, to
+the mobile OS server, for display as a push notification
+of the client.
```

Comment:
What is the expected behaviour for the client in case it receives a push notification? It gets a fraction of a second of compute time on the mobile device. Should it start and receive the payment. Should it show the notification and receive the payment when the user clicks on it?

We might need inputs from mobile devs here as the compute time is really a limiting factor.

Reply:
Hmm @jurvis @jesseposner ?

Reply:
Maybe we can just say

```Markdown
The client SHOULD process what is needed to
react to the notification (e.g. connect to the LSP and
accept the payment for `reason` code `1`) if the mobile
environment allows it.
```

Reply:
Done.